src/app/utils/avatar/avatarUtils.ts

```diff
@@ -0,0 +1,26 @@
+function getAvatarExpiration(url: string): Date | null {
+  const dateRegex = /X-Amz-Date=(\d{8}T\d{6})Z/;
+  const expiresRegex = /X-Amz-Expires=(\d+)/;
+
+  const dateMatch = dateRegex.exec(url);
+  const expiresMatch = expiresRegex.exec(url);
+
```

Comment:
will define this more generically, as it is now very much coupled to aws. It could be added as configuration parameters to check for expiration